### PR TITLE
Use upstream `wasmedge-sys`

### LINKED ISSUE #7: Remove `wasmedge_sys_ffi` and use upstream `wasmedge-sys::ffi`
The package `wasmedge_sys_ffi` is doing exactly the same as the upstream `wasmedge-sys::ffi`, but `wasmedge-sys` has more features, like the `standalone` feature.
There doesn't seem to be any particular value added by `wasmedge_sys_ffi`.

### PR DESCRIPTION
**Pull request by jprendes**
Fixes #7

### CODE REVIEW

File: src/sdk/plugin.rs
Comment (L-jasmine) [inline src/sdk/plugin.rs:1]: This file conflicts with the previous #8. I think #8 is better, so please update it.

File: src/core/module.rs [hunk 1]
```diff
@@ -307,3 +307,14 @@ impl<T: AsInnerInstance> AsInstance for T {
...
+trait TryAsString {
+    fn try_as_string(&self) -> Result<String, std::str::Utf8Error>;
+}
+
+impl TryAsString for ffi::WasmEdge_String {
+    fn try_as_string(&self) -> Result<String, std::str::Utf8Error> {
+        let cstr = unsafe { std::ffi::CStr::from_ptr(self.Buf as *const _) };
+        Ok(cstr.to_str()?.to_string())
+    }
+}
```
Comment (L-jasmine) [inline src/core/module.rs:320]: Can use std traits in Rust to replace it?

Comment (jprendes) [reply src/core/module.rs:320]: No, in this crate `ffi::WasmEdge_String` is a foreign type belonging to a different crate, and any std trait would be a foreign  trait.
We can't define foreign traits on foreign types, at least one of them has to be defined locally.

A trait from std could be used if we were in the `wasmedge-sys` crate.

Comment (L-jasmine) [reply src/core/module.rs:320]: ok, I will impl std trait in wasmedge-sys.

Comment (jprendes) [reply src/core/module.rs:320]: I think the trait should be [`TryInto`](https://doc.rust-lang.org/std/convert/trait.TryInto.html) (as opposed to `Into`  that's currently implemented in `wasmedge_sys_ffi`)

**Comment by L-jasmine** [comment]
@jprendes Please resolve this conflict and I will merge this pr. Thanks

**Comment by jprendes** [comment]
Rebased onto latest `main`.
Please note this is still using the older `wasmedge-sys` `0.17.5`, as `0.19.1` introduces a major API change.

**Pull request merged**